test/core/event_engine/event_engine_test_utils.cc

```diff
@@ -75,7 +75,7 @@ std::string GetNextSendMessage() {
   return tmp_s;
 }
 
-void WaitForSingleOwner(std::shared_ptr<EventEngine>&& engine) {
+void WaitForSingleOwner(std::shared_ptr<EventEngine> engine) {
```

Comment:
Why get rid of the move-only requirement? If it's passed by copy, this method will hang forever (use_count > 1 in a blocking call).

Reply:
It's still possible to pass a reference through here without moving (or something) - I got the equivalent problem with the construction in end2end_tests.cc.

I'd say this works and if you want to tinker later then let's do that later.

Reply:
This could use a caution added to the `WaitForSingleOwner` documentation, then, since the most natural usage is wrong. `Callers must give up their ref, or this method will block forever. Usage: WaitForSingleOwner(std::move(engine))`.

Reply:
done